subsys/settings/settings_store.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2018 Laczen
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ * Copyright (c) 2015 Runtime Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <stdio.h>
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <sys/types.h>
+#include <errno.h>
+#include <misc/__assert.h>
+
+#include <settings/settings.h>
+#include "settings_priv.h"
+
+sys_slist_t  settings_load_srcs;
+struct settings_store *settings_save_dst;
+
+void settings_src_register(struct settings_store *cs)
+{
+	sys_snode_t *prev, *cur;
+
+	prev = NULL;
+
+	SYS_SLIST_FOR_EACH_NODE(&settings_load_srcs, cur) {
+		prev = cur;
+	}
+
+	sys_slist_insert(&settings_load_srcs, prev, &cs->cs_next);
+}
+
+void settings_dst_register(struct settings_store *cs)
+{
+	settings_save_dst = cs;
+}
+
+int settings_load(void)
+{
+	struct settings_store *cs;
+
+	/*
+	 * for every settings source
+	 *    load config
+	 *    apply config
+	 */
+	SYS_SLIST_FOR_EACH_CONTAINER(&settings_load_srcs, cs, cs_next) {
+		cs->cs_itf->csi_load(cs);
+	}
+
+	/*
+	 * finally commit the settings
+	 */
+	return settings_commit(NULL);
+}
+
+
+/*
+ * Append a single value to persisted config.
+ */
+int settings_save_one(const char *name, void *value, size_t val_len)
+{
+	struct settings_store *cs;
+
+	cs = settings_save_dst;
+	if (!cs) {
+		return -ENOENT;
+	}
+	return cs->cs_itf->csi_save(cs, name, (char *)value, val_len);
```

Comment:
a couple of checks could be good before saving:
- strlen(name) against SETTINGS_MAX_NAME_LEN
- val_len against SETTINGS_MAX_VAL_LEN

not sure if similar checks should be done on the backends after loading...